src/DomainNetMapImp.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef DOMAINNETMAPIMP_HPP_INCLUDE
+#define DOMAINNETMAPIMP_HPP_INCLUDE
+
+#include "DomainNetMap.hpp"
+
+#include <memory>
+
+#include "geopm/json11.hpp"
+
+#include "DenseLayer.hpp"
+#include "LocalNeuralNet.hpp"
+#include "TensorOneD.hpp"
+#include "TensorTwoD.hpp"
+#include "NNFactory.hpp"
+
+namespace geopm
+{
+    class PlatformIO;
+
+    class DomainNetMapImp : public DomainNetMap
+    {
+        public:
+            DomainNetMapImp(const std::string &nn_path, geopm_domain_e domain_type,
+                            int domain_index);
+
+            DomainNetMapImp(const std::string &nn_path, geopm_domain_e domain_type,
+                            int domain_index, PlatformIO &plat_io,
+                            std::shared_ptr<NNFactory> nn_factory);
+
+            void sample() override;
+            /// @brief Generates the names for trace columns from the appropriate field in the neural net.
+            //         In this case, region classification names annotated with domain type and index. 
+            std::vector<std::string> trace_names() const override;
+            /// @brief Populates trace values from last_output for each index within each domain type.
+            ///        In this case, region classification logits. 
+            /// @return Returns a vector of doubles containing trace values
+            std::vector<double> trace_values() const override;
+            /// @brief Populates a map of trace names to the latest output from the neural net.
+            ///        In this case, region classification names to their respective logits.
+            ///
+            /// @return A map of string, double containing region class and logits.
+            std::map<std::string, double> last_output() const override;
+
+        private:
+            std::shared_ptr<DenseLayer> json_to_DenseLayer(const json11::Json &obj) const;
+            TensorOneD json_to_TensorOneD(const json11::Json &obj) const;
+            TensorTwoD json_to_TensorTwoD(const json11::Json &obj) const;
+
+            PlatformIO &m_platform_io;
+            std::shared_ptr<NNFactory> m_nn_factory;
+
+            struct m_signal_s
+            {
+                int batch_idx;
+                double signal;
+            };
+
+            struct m_delta_signal_s
+            {
+                int batch_idx_num;
+                int batch_idx_den;
+                double signal_num;
+                double signal_den;
+                double signal_num_last;
+                double signal_den_last;
+            };
+
+            static const std::vector<std::string> M_EXPECTED_KEYS;
+            static constexpr int M_MAX_NNET_SIZE_B = 1024 * 1024;
```

Comment:
Why SIZE_B?  Is there a SIZE_A?
This could be more descriptive.

Reply:
B = Bytes :)
I added this as a result of feedback in our review meeting. The request was to specify the units. Should we spell out BYTES or should we simply put it in a comment?

Reply:
Ah, thanks for the explanation.  I'd put it in a comment:
```suggestion
            static constexpr int M_MAX_NNET_SIZE = 1024 * 1024; // In bytes
```